index.bs

```diff
@@ -361,7 +361,70 @@ Report long tasks {#report-long-tasks}
+it could make existing privacy attacks faster. Mitigations for this are possible and can be implemented
+as needed:
+
+* Further clamp the long task duration provided by the API to make attacks harder to exploit.
+* Limit the number of origins for which longtasks are exposed by the API, and obfuscate the attribution of
+    any tasks afterwards.
+* Allow dropping the culprit/attribution information after a certain threshold.
+* Add a built-in delay to the timing information exposed to make attacks dependent on longtask volume harder
+    to execute.
+
+What is Exposed to Observers? {#what-is-exposed}
+--------------------------------------------------------
+
+All observers within the top level page (i.e. all iframes in the page and the main frame) will receive
+notifications about presence of long tasks. We expose the start time of the task, its duration, and a
+pointer to the culprit frame. This information can already be observed today, and with higher resolution,
+using setTimeout. Observers in other different pages (tabs or windows) should not receive notifications,
+regardless of the architecture of the user agent.
+
+Cross origin rules for what is exposed:
+* Cross-origin observers may see the direction of the culprit e.g if the culprit is a deeply nested iframe,
+    then the host page can see the first cross-origin between itself and the culprit.
+* Conversely, if the culprit is the top level page, then a deeply embedded iframe can see that a longtask
+    occurrred in its cross-origin ancestor but does not receive any information about it.
+
+Opt-in: In a future version of this API, sites could opt into revealing their attribution info to trusted
+cross origins using a mechanism like the "Timing-Allow-Origin" header. This is very useful for enabling third
+party content providers to share their attribution info with host sites, as they often partner to tackle
+performance issues.
+
+Attack Scenarios Considered {#attack-scenarios}
+--------------------------------------------------------
+
+The following are the timing attacks considered:
+
+1. <b>Timing attacks</b>, of which two variants are considered:
+
+    1. <b>Traditional timing attacks</b>: using external resource load time to reveal the size of
+        private data. For instance the number of hidden pictures in a gallery, whether username is
+        valid, etc. See an <a href="http://crypto.stanford.edu/~dabo/papers/webtiming.pdf">example</a>.
+
+    1. <b>Side-channel timing attacks</b>: using time for video parsing, script parsing, App Cache reads
+        or Cache API (service workers) usage to uniquely identify a user, or to create a profile of the
+        user’s age, gender, location, and interests etc. For
+        <a href="https://tom.vg/papers/timing-attacks_ccs2015.pdf">instance</a>, status updates from
+        Facebook Pages can be limited to certain demographic (eg. females of age 20-30) the file size of
```

Comment:
I don't know if we want to call out Facebook specifically here. Maybe replace with "social network"?

Reply:
Done